From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martin Wilck <mwilck@suse.com>
Date: Thu, 27 Oct 2022 16:53:04 +0200
Subject: [PATCH] multipath-tools: Makefile.inc: untangle paths and source
 directories

Only the installation paths can be customized. Move the definitions
of $(multipathdir) etc. further down.

Signed-off-by: Martin Wilck <mwilck@suse.com>
Signed-off-by: Benjamin Marzinski <bmarzins@redhat.com>
---
 Makefile.inc | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/Makefile.inc b/Makefile.inc
index 1a08b8fa..6ec8201b 100644
--- a/Makefile.inc
+++ b/Makefile.inc
@@ -31,8 +31,6 @@ endif
 SYSTEMD := $(strip $(or $(shell $(PKGCONFIG) --modversion libsystemd 2>/dev/null | awk '{print $$1}'), \
 			$(shell systemctl --version 2>/dev/null | sed -n 's/systemd \([0-9]*\).*/\1/p')))
 
-# Allow user to override default location.
-
 # Paths. All these can be overridden on the "make" command line.
 prefix		=
 # Prefix for binaries
@@ -49,20 +47,11 @@ modulesloaddir	:= $(systemd_prefix)/lib/modules-load.d
 libudevdir	:= $(systemd_prefix)/lib/udev
 udevrulesdir	:= $(libudevdir)/rules.d
 bindir		= $(exec_prefix)/sbin
-multipathdir	= $(TOPDIR)/libmultipath
-daemondir       = $(TOPDIR)/multipathd
-mpathutildir	= $(TOPDIR)/libmpathutil
 mandir		:= $(usr_prefix)/share/man
 LIB		:= $(if $(shell test -d /lib64 && echo 1),lib64,lib)
 syslibdir	= $(prefix)/$(LIB)
 usrlibdir	= $(usr_prefix)/$(LIB)
 libdir		= $(prefix)/$(LIB)/multipath
-mpathpersistdir	= $(TOPDIR)/libmpathpersist
-mpathcmddir	= $(TOPDIR)/libmpathcmd
-mpathvaliddir	= $(TOPDIR)/libmpathvalid
-thirdpartydir	= $(TOPDIR)/third-party
-libdmmpdir	= $(TOPDIR)/libdmmp
-nvmedir		= $(TOPDIR)/libmultipath/nvme
 includedir	= $(usr_prefix)/include
 pkgconfdir	= $(usrlibdir)/pkgconfig
 plugindir       := $(prefix)/$(LIB)/multipath
@@ -124,6 +113,18 @@ SHARED_FLAGS	= -shared
 LDFLAGS		:= $(LDFLAGS) -Wl,-z,relro -Wl,-z,now -Wl,-z,defs
 BIN_LDFLAGS	= -pie
 
+# Source code directories. Don't modify.
+
+multipathdir	= $(TOPDIR)/libmultipath
+daemondir       = $(TOPDIR)/multipathd
+mpathutildir	= $(TOPDIR)/libmpathutil
+mpathpersistdir	= $(TOPDIR)/libmpathpersist
+mpathcmddir	= $(TOPDIR)/libmpathcmd
+mpathvaliddir	= $(TOPDIR)/libmpathvalid
+thirdpartydir	= $(TOPDIR)/third-party
+libdmmpdir	= $(TOPDIR)/libdmmp
+nvmedir		= $(TOPDIR)/libmultipath/nvme
+
 # Check whether a function with name $1 has been declared in header file $2.
 check_func = $(shell \
 	if grep -Eq "^[^[:blank:]]+[[:blank:]]+$1[[:blank:]]*(.*)*" "$2"; then \
